tests/migrations/test_optimizer.py

```diff
@@ -830,3 +830,30 @@ def test_optimize_elidable_operation(self):
                 migrations.CreateModel("Phou", [("name", models.CharField(max_length=255))]),
             ],
         )
+
+    def test_optimizer_crashes_when_fields_names_are_swapped(self):
+        self.assertOptimizesTo(
+            [
+                migrations.CreateModel(
+                    'MyModel',
+                    [
+                        ('field_a', models.IntegerField()),
+                        ('field_b', models.IntegerField()),
+                    ],
+                ),
+                migrations.RunPython(migrations.RunPython.noop),
+                migrations.RenameField('MyModel', 'field_a', 'field_c'),
+                migrations.RenameField('MyModel', 'field_b', 'field_a'),
+                migrations.RenameField('MyModel', 'field_c', 'field_b'),
+            ],
+            [
+                migrations.CreateModel(
+                    'MyModel',
+                    [
+                        ('field_b', models.IntegerField()),
+                        ('field_a', models.IntegerField()),
+                    ],
+                ),
+                migrations.RunPython(migrations.RunPython.noop),
+            ],
```

Comment:
This optimization can't be allowed.

What if `RunPython` pointed at a function that referenced specific fields? e.g.


```python
def forwards(apps, schema_editor):
    apps.get_model('app.MyModel').objects.filter(field_a__lte=42).update(field_a=14)

operations = [
    migrations.CreateModel(
        'MyModel',
        [
            ('field_a', models.IntegerField()),
            ('field_b', models.IntegerField()),
        ],
    ),
    migrations.RunPython(forwards),
    migrations.RenameField('MyModel', 'field_a', 'field_c'),
]
```

`RunPython` opaque operations that must be optimization barriers as they can refer to model definitions at any time. They require either to be marked as `elidable` if they can be removed or adjusted by users on migration squashing.